File: C#/1492. The kth Factor of n.cs
public class Solution {
    public int KthFactor(int n, int k) {
        List<int> list = new List<int>();
        for(int i = 1; i<=n; i++){
            if(n%i==0){
                list.Add(i);
            }
        }
        if(k>list.Count){
            return -1;
        }
        return list[k-1];
    }
}

File: C++/1492. The kth Factor of n.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        vector<int> vec;
        for(int i = 1; i<=n; i++){
            if(n%i==0){
                vec.push_back(i);
            }
        }
        if(k>vec.size()){
            return -1;
        }
        return vec[k-1];
    }
};
